modules/unis.md

```diff
@@ -0,0 +1,153 @@
+   - `x=1` = output grounded.
+* `0b 0100 xxxx`: flicker output. `xxxx`:
+   - 1: 1 Hz
+   - 2: 2 Hz
+   - 3: 3 Hz
+   - 4: 4 Hz
+   - 5: 5 Hz
+   - 6: 10 Hz
+   - 7: 33 tick/min
+   - 8: 66 tick/min
+
+## Configuration
+
+Configuration consists of 55 bytes. First `24` bytes are same as in MTB-UNI module.
+
+1. 16 bytes of safe outputs state (outputs indexed in order 0 to 15).
+2. 8 bytes of input keep delay.
+   - Byte 0: `0bBBBBAAAA`. `A` = delay of input 0, `B` = delay of input 1.
+   - Byte 1: delay of input 2, delay of input 3.
+   - ...
+   Each input has 16 values of input delay. `0`=0.0s, `1`=0.1s, `2`=0.2s, ...,
+   `15`=1.5s.
+3. 1 byte mask, which servo outputs are active
+   - `0b00654321`, 1 = output active, 0 = output disabled
+4. 24 bytes for servo positions
+   - 2 byte value for servo 1 position 1
+   - 2 byte value for servo 1 position 2
+   - 2 byte value for servo 2 position 1
+   - ...
+5. 6 bytes for servo speed
+   - speed for servo 1
+   - speed for servo 2
+   - ...
+   
+When input goes to logical 0, it must remain in this state for *input keep
+delay* to consider the input as logical 0. Only after the time input changed
+event is reported to master board.
+
+## Module-specific commands
+
+Master → slave:
+
+* *Set Configuration*: n.o. data bytes: 55. Whole configuration is sent.
+* *Get Configuration*: n.o. data bytes: 0.
+* *Get Input*: n.o. data bytes: 0.
+* *Set Output*: n.o. data bytes: variable.
+  - State of all outputs is always sent.
+  - Data byte 0: full state mask for outputs `0b00fedcba`.
+  - Data byte 1: full state mask for outputs `0bFEDCBA98`.
+  - Data byte 2: full state mask for outputs `0b76543210`.
+  - Data byte 3: binary state of outputs `0b00fedcba`.
+  - Data byte 4: binary state of outputs `0bFEDCBA98`.
+  - Data byte 5: binary state of outputs `0b76543210`.
+  - Data byte 6–n: state of full-state outputs in order 0–F.
+  - Examples (only data bytes are written):
+    - Set output 1 as active, other as inactive: `0x00 0x00 0x00 0x02`.
+    - Set output 8 to S-COM code `0x0A`, output 10 as flickering with period
+      2 Hz, output 15 as active, other as inactive:
+      `0x05 0x00 0xF0 0x00 0x42 0x8A`.
+  - Bits in *binary state output* which are masked by *full state mask* can
+    contain any value. This value is ignored.
+* *Firmware Write Flash*
+  - Data byte 0: first byte flash address high.
+  - Data byte 1: first byte flash address low.
+  - Data byte 2–65: 64 bytes of memory data.
+* *Module-specific command*
+  - Data byte 0 = `0x01` = set servo position.
+    - Data byte 1 = position identification `0b0nnn000p` where p = position, n = servo number
+    - Data byte 2 + byte 3 = new position for servo
+    (p = position  - 0=1st position, 1=2nd position, n = servo number - possible values 1-6)
+  - Data byte 0 = `0x02` = set servo speed.
+    - Data byte 1 = position identification `0b0nnn000p` where p = position, n = servo number
+    - Data byte 2 + byte 3: new speed for servo.
```

Comment:
Opět by se hodilo popsat, co přesně je to `servo speed` - povolený rozsah. Z řádků 51-53 mi vyplývá, že `servo speed` má rozsah 1 byte, ale tady se uvádí 2 byty. Jak je to?